tutorials/eth-api/thirdweb.md

```diff
@@ -0,0 +1,285 @@
+        pnpm install
+        ```
+
+
+Next, you'll need to create a client ID for your thirdweb project and specify it in a `.env` file. 
+
+## Specify Client ID {: #specify-client-id }
+
+Before you launch your dApp (locally or publicly deployed), you must have a thirdweb Client ID associated with your project. A thirdweb Client ID is synonymous with an API key. You can create a free API key by [signing into your thirdweb Account and navigating to **Settings** then click on **API Keys**](https://thirdweb.com/dashboard/settings/api-keys){target=\_blank}.
+
+Press **Create API Key** then take the following steps:
+
+1. Give your API key a name
+2. Enter the allowed domains that the API key should accept requests from. It's recommended that you allow only necessary domains, but for development purposes, you can select **Allow all domains**
+3. Press **Next** and confirm the prompt on the next page
+
+![thirdweb create API key](/images/tutorials/eth-api/thirdweb/thirdweb-1.webp "Creating an API key on thirdweb")
+
+Now, create a file a called `.env.local` at the root level directory of your project. Add your client ID as follows:
+
+```bash
+NEXT_PUBLIC_TW_CLIENT_ID="INSERT-CLIENT-ID"
+```
+
+There are no other fields that you need to specify in your `.env` file at this time. If you're uncertain about formatting, you can refer to the `.env.example` file included in the project.
+
+## Run the project {: #run-the-project }
+
+We're not finished with the project yet, but it should be at a stage where you can launch the template and see the web app load successfully in your browser. 
+
+=== "npm"
+
+    ```bash
+    npm run dev
+    ```
+
+=== "yarn"
+
+    ```bash
+    yarn dev
+    ```
+
+=== "pnpm"
+
+    ```bash
+    pnpm dev
+    ```
+
+If you see a blank screen, this typically means that you've failed to properly configure your client ID. 
+
+## Add Support for Moonbase Alpha {: #add-support-for-moonbase-alpha }
+
+thirdweb offers a small number of chains from `@thirdweb/chains` and does not include Moonbeam networks in that list, so you'll need to specify the network details including chain ID and RPC URL. You can create a custom chain with [`defineChain`](https://portal.thirdweb.com/references/typescript/v5/defineChain){target=\_blank} as follows:
+
+=== "Moonbeam"
+
+    ```typescript
+    import { defineChain } from 'thirdweb';
+    const moonbase = defineChain({
+    id: {{ networks.moonbeam.chain_id }},
+    rpc: '{{ networks.moonbeam.public_rpc_url }}',
+    })
+    ```
+
+=== "Moonriver"
+
+    ```typescript
+    import { defineChain } from 'thirdweb';
+    const moonbase = defineChain({
+    id: {{ networks.moonriver.chain_id }},
+    rpc: '{{ networks.moonriver.public_rpc_url }}',
+    })
+    ```
+
+=== "Moonbase Alpha"
+
+    ```typescript
+    import { defineChain } from 'thirdweb';
+    const moonbase = defineChain({
+    id: {{ networks.moonbase.chain_id }},
+    rpc: '{{ networks.moonbase.rpc_url }}',
+    })
+    ```
```

Comment:
where we adding this? a title with the file name would be helpful here. and these snippets need to be formatted